children/tests/test_api.py

```diff
@@ -790,6 +793,69 @@ def test_child_enrolment_count(
     assert executed["data"]["child"]["enrolmentCount"] == expected_count
 
 
+@freeze_time("2021-02-02T12:00:00Z")
+@pytest.mark.parametrize(
+    "year_delta,expected_count",
+    [(-2, 0), (-1, 1), (0, 2), (None, 2), (1, 1), (2, 0)],
+)
+def test_enrolment_count_with_ticket_system_passwords(
+    guardian_api_client, child_with_user_guardian, year_delta, expected_count
+):
+    variables = {"id": to_global_id("ChildNode", child_with_user_guardian.id)}
+    if year_delta:
+        variables["year"] = timezone.now().year + year_delta
+
+    last_year = timezone.now() - timedelta(days=365)
+    this_year = timezone.now() + timedelta(days=1)
+    next_year = timezone.now() + timedelta(days=365)
```

Comment:
This handles leap years correctly (There aren't always 365 days in a year) and fixes this_year (no need for +1 day): 
```python
...
from dateutil.relativedelta import relativedelta
...
    last_year = timezone.now() - relativedelta(years=1)
    this_year = timezone.now()
    next_year = timezone.now() + relativedelta(years=1)
```

Reply:
Taken into account.